sw/device/silicon_creator/rom/rom.c

```diff
@@ -379,6 +399,34 @@ static void rom_pre_boot_check(void) {
   CFI_FUNC_COUNTER_INCREMENT(rom_counters, kCfiRomPreBootCheck, 7);
 }
 
+/**
+ * Measures the software readable OTP partitions.
+ *
+ * @param measurement Pointer to the measurement of the partitions.
+ * @return rom_error_t Result of the operation.
+ */
+static rom_error_t rom_measure_otp_partitions(
+    keymgr_binding_value_t *measurement) {
+  memset(measurement, (int)rnd_uint32(), sizeof(keymgr_binding_value_t));
+  hmac_sha256_init();
+  for (size_t p = 0; p < kOtpPartitionNumPartitions; ++p) {
+    // Rather than calling `hmac_sha256_update()` update here, we copy the OTP
+    // data directly into the HMAC FIFO to reduce code size.
+    HARDENED_CHECK_LE(kOtpPartitions[p].size / sizeof(uint32_t),
+                      HMAC_MSG_FIFO_SIZE_WORDS);
+    HARDENED_CHECK_EQ(kOtpPartitions[p].size % sizeof(uint32_t), 0);
+    otp_dai_read(
```

Comment:
I guess the FIFO window is large enough but is it worth adding a check against the partition size?

Reply:
I do, that is what `HARDENED_CHECK_LE(kOtpPartitions[p].size / sizeof(uint32_t), HMAC_MSG_FIFO_SIZE_WORDS);` `HMAC_MSG_FIFO_SIZE_WORDS` is autogenerated from the fifo window size: https://opentitan.org/book/hw/ip/hmac/doc/registers.html#msg_fifo